Azure/azure-cli
# [App Service] `az functionapp create`: Enable distributed tracing for non consumption apps

**Related command**
`az functionapp create`

**Description**
Currently, in the Azure Portal, full monitoring is enabled by default for non consumption Java function apps that will have deployed code (as opposed to container image). We need to have the same default behavior in the Azure CLI. This is achieved by setting the `APPLICATIONINSIGHTS_ENABLE_AGENT` app setting to true.

**Testing Guide**
1. Create function app with `az functionapp create -g {} -n {} -s {} --plan {} --runtime java --functions-version 4` command
2. Check that `APPLICATIONINSIGHTS_ENABLE_AGENT` is set to true with `az functionapp config apsettings list -g {} -n {}`

**Relevant Documentation**
https://learn.microsoft.com/en-us/azure/azure-monitor/app/monitor-functions#distributed-tracing-for-java-applications-public-preview

---

This checklist is used to make sure that common guidelines for a pull request are followed.
